### DISCUSSION TITLE
External YAML Configuration File

### DISCUSSION THREAD

**Discussion by brachycera** [gickowtf/pixoo-homeassistant#84] [Q&A]
Hello,

thanks for the nice pixoo HA integration!

Is there a possibility to use an external yaml file for the configuration e.g. `/config/configuration.yaml`? It's a little cumbersome to make changes to the configuration, save, test, open the configuration again. I would love to edit the pixoo configuration in VS-Code 😉

**Comment by SB2DD** [comment]
Unfortunately, there currently isn't an easy way to do so, but this is a nice idea. Although I'm not against this idea, my only problem is that I'm unsure as to what the advantages would be like. Like, you'd still kind of need to like reload the config entry every time you wanted to apply some changes, so I don't really see the advantages of using an external file anyways...

**Reply by brachycera** [reply]
> my only problem is that I'm unsure as to what the advantages

I could use the code suggestion from the VS-Code Extension https://marketplace.visualstudio.com/items?itemName=keesschollaart.vscode-home-assistant. Or the undo/redo function of the IDE 😉 

> you'd still kind of need to like reload the config entry every time you wanted to apply some changes

That's true I didn't thought about that. I guess this should have been handled in the integration. Maybe checking for file changes (date, hash) and then reload the config?! 🤷 

Thanks for your input @Mrredstone5230
